homeassistant/components/mqtt/binary_sensor.py

```diff
@@ -224,6 +226,20 @@ def state_message_received(msg):
                 }
             },
         )
+        if (
+            (expire_after := self._config.get(CONF_EXPIRE_AFTER)) is not None
+            and expire_after > 0
+            and (last_state := await self.async_get_last_state()) is not None
+            and last_state != STATE_UNAVAILABLE
+        ):
+            self._expired = False
+            self._state = last_state.state
+            expiration_at = dt_util.utcnow() + timedelta(seconds=expire_after)
+            self._expiration_trigger = async_track_point_in_utc_time(
```

Comment:
I think we need to call make sure the expiration triggers are cleaned up by cancelling them from `self.async_will_remove_from_hass`.

Reply:
That is what is happening. This code part is for the restore operation.

Reply:
Where does it happen?